sdks/python/apache_beam/examples/wordcount_it_test.py

```diff
@@ -31,7 +31,7 @@ class WordCountIT(unittest.TestCase):
   @attr('IT')
   def test_wordcount_it(self):
     # Set extra options to the pipeline for test purpose
-    extra_opts = {'on_success_matcher': PipelineStateMatcher()}
```

Comment:
good to learn this pattern. 

However, I'm not sure how to use them for matchers. According to [doc](https://docs.python.org/2/library/functions.html#all), `any` and `all` check`True`/`False` elements, but for matcher(s) we need to call `assert_that` for verification.